fix(binaryfile): avoid numpy deprecation warnings

w-bonelli opened the pull request:
In the future, [numpy will stop treating arrays with 1 element and dimension >0 as scalars](https://numpy.org/doc/stable/release/1.25.0-notes.html#deprecations). Use `.item()` to access the single element.

This may need to be fixed elsewhere in the codebase, this is the only place I have seen warnings from recently.

w-bonelli commented:
I took a brief look through the repo and the [warnings raised by autotests](https://github.com/modflowpy/flopy/actions/runs/5523798037/jobs/10075279695#step:11:1958) and don't see this anywhere else. Will keep an eye out for more though

mwtoews commented:
Thanks, I was considering addressing this new warning, you beat me to it! And agree that using `.item()` is a good approach. Also, keep in mind that new NumPy releases only support Python 3.9 or later, so new deprecation warnings are not visible with Python 3.8 runs.

Missed [one](https://github.com/modflowpy/flopy/actions/runs/5502252129/jobs/10026405737?pr=1868#step:11:2218) from a different package, however:
> test_gridutil.py: 7442 warnings
  /home/runner/work/flopy/flopy/flopy/utils/gridutil.py:91: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    top[n] = tp

w-bonelli commented:
Thanks @mwtoews I did not realize this

> new NumPy releases only support Python 3.9 or later, so new deprecation warnings are not visible with Python 3.8 runs

Fixed the one in gridutil and added type hints to the ndarray params in the `get_disu_kwargs()` signature
